### DISCUSSION TITLE
Set value in cffi::Struct

### DISCUSSION THREAD

**Discussion by nico-robert** [apnadkarni/tcl-cffi#132] [General]
Hello @apnadkarni ,
I try to do this structure ([PIXELFORMATDESCRIPTOR](https://docs.microsoft.com/en-us/windows/win32/api/wingdi/ns-wingdi-pixelformatdescriptor)) to be able to set `dwFlags` (type _DWORD_).
For this field my value should be like this C example :
`pfd.dwFlags = PFD_DRAW_TO_WINDOW | PFD_SUPPORT_OPENGL | PFD_DOUBLEBUFFER;`

For the moment I have not implemented my constant values but I know the value for each :

- PFD_DRAW_TO_WINDOW  = 0x00000004
- PFD_SUPPORT_OPENGL     = 0x00000020
- PFD_DOUBLEBUFFER          = 0x00000001

My goal will be to do like pseudo code below, but I don’t know how to assign this syntax 
```
set pfd [PIXELFORMATDESCRIPTOR allocate]
PIXELFORMATDESCRIPTOR setnative $pfd dwFlags {0x00000004 | 0x00000020 | 0x00000001}
```
Regards
Nicolas

**Comment by nico-robert** [comment]
I believe this solution works :
`PIXELFORMATDESCRIPTOR setnative $pfd dwFlags [expr {0x00000004 | 0x00000020 | 0x00000001}]`

**Comment by apnadkarni** [comment]
Yes, you can always use any integer, computed with expr for storing integer values. Alternatively, if you define an `enum` with the appropriate constants and the `bitmask` attribute, you can also pass it as directly as

```
PIXELFORMATDESCRIPTOR setnative $pfd dwFlags {PFD_DRAW_TO_WINDOW PFD_SUPPORT_OPENGL  PFD_DOUBLEBUFFER}
```

**Reply by nico-robert** [reply]
I like your method , it's more clever.
Thank you
